### PR TITLE
Gilad/jwt authentication

### PR DESCRIPTION
**Pull request by gilRoz23**

### CODE REVIEW

File: src/main/kotlin/com/hibob/kotlinEx/kotlinBasics.kt
Comment (elijulian) [inline src/main/kotlin/com/hibob/kotlinEx/kotlinBasics.kt:1]: this should be in a separate pr

File: src/main/kotlin/com/hibob/academy/resource/ExampleResource.kt
Comment (elijulian) [inline src/main/kotlin/com/hibob/academy/resource/ExampleResource.kt:1]: this should be a separate pr

File: src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt
Comment (elijulian) [inline src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt:1]: why commented out?

Comment (elijulian) [reply src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt:1]: unless you want to do this in a separate pr, which i think makes sense

File: src/main/kotlin/com/hibob/academy/resource/sessionResource.kt
Comment (elijulian) [inline src/main/kotlin/com/hibob/academy/resource/sessionResource.kt:1]: this should be in the resource package

Comment (gilRoz23) [reply src/main/kotlin/com/hibob/academy/resource/sessionResource.kt:1]: it is. do you mean in resources in other folder maybe?

File: src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+
+@Provider
+@Component
+class AuthenticationFilter(private val status: HttpCodeStatusMapper) : ContainerRequestFilter {
+    @Throws(Nothing::class)
+    override fun filter(requestContext: ContainerRequestContext) {
+
+        if (requestContext.uriInfo.path == "api/gilad/session/login") return
+        else {
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt:22]: if you return on the previous line, no need for the else.

File: src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            val cookies = requestContext.cookies
+            val cookieVal = cookies.get("cookieVal")?.value.toString()
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt:24]: `"cookieVal"` should be a const, and reused here and in the resource file

File: src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+            val retVal: Jws<Claims>? = verify(cookieVal)
+
+            if (retVal == null) {
+                requestContext.abortWith(Response.status(Response.Status.UNAUTHORIZED).build())
+            }
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt:30]: this should be handled in the catch clause

File: src/main/kotlin/com/hibob/academy/resource/sessionResource.kt
Comment (elijulian) [reply src/main/kotlin/com/hibob/academy/resource/sessionResource.kt:1]: i mean that it should be moved to `com.hibob.academy.resource`

File: src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt (hunk 3 above)
Comment (gilRoz23) [reply src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt:31]: changed it in the filter func. is that what you meant?

**Review by elijulian — APPROVED** [review]
